test_runner/fixtures/neon_fixtures.py

```diff
@@ -2842,6 +2843,33 @@ def http_query(self, query, args, **kwargs):
             assert response.status_code == kwargs["expected_code"], f"response: {response.json()}"
         return response.json()
 
+    async def http2_query(self, query, args, **kwargs):
+        # TODO maybe use default values if not provided
+        user = kwargs["user"]
+        password = kwargs["password"]
+        expected_code = kwargs.get("expected_code")
+
+        connstr = f"postgresql://{user}:{password}@{self.domain}:{self.proxy_port}/postgres"
+        async with httpx.AsyncClient(
+            http2=True, verify=str(self.test_output_dir / "proxy.crt")
+        ) as client:
+            response = await client.post(
+                f"https://{self.domain}:{self.external_http_port}/sql",
+                json={"query": query, "params": args},
+                headers={
+                    "Content-Type": "application/sql",
+                    "Neon-Connection-String": connstr,
+                    "Neon-Pool-Opt-In": "true",
+                },
+            )
+            assert response.http_version == "HTTP/2"
+
+            if expected_code is not None:
+                assert (
+                    response.status_code == kwargs["expected_code"]
```

Comment:
NIT: you already got the `expected_code` from `kwargs`

```python
response.status_code == expected_code
```

or, you can do this way

```python
if (expected_code := kwargs.get("expected_code")) is not None:
   assert response.status_code == expected_code
```

and remove the line above

```python
expected_code = kwargs.get("expected_code")
```

Reply:
Ah, I copied the code from above which does the same